### PR TITLE
feat: Rename LoginAction to LoginSuccessAction

### LINKED ISSUE #135: setUser after login in SDK
^ title

### PR DESCRIPTION
**Pull request by HazAT**
Fixes #135

### CODE REVIEW

File: lib/redux/reducers.dart [hunk 1]
```diff
@@ -15,7 +15,7 @@ AppState appReducer(AppState state, dynamic action) =>
 final globalReducer = combineReducers<GlobalState>([
   TypedReducer<GlobalState, SwitchTabAction>(_switchTabAction),
   TypedReducer<GlobalState, RehydrateSuccessAction>(_rehydrateSuccessAction),
-  TypedReducer<GlobalState, LoginAction>(_loginAction),
+  TypedReducer<GlobalState, LoginSuccessAction>(_loginAction),
```
Comment (denrase) [inline lib/redux/reducers.dart:18]: We should also rename the handling method from `_loginAction ` to `_loginSuccessAction `

File: lib/redux/middleware/sentry_sdk_middleware.dart [hunk 2]
```diff
@@ -11,6 +11,12 @@ class SentrySdkMiddleware extends MiddlewareClass<AppState> {
     if (action is ApiFailureAction) {
       Sentry.captureException(action.error, stackTrace: action.stackTrace);
     }
+    if (action is FetchAuthenticatedUserSuccessAction) {
+      Sentry.configureScope((scope) => scope.user = User(email: action.me.email, id: action.me.id, ipAddress: "{{auto}}"));
```
Comment (denrase) [inline lib/redux/middleware/sentry_sdk_middleware.dart]: Why do we need the `ipAddress: "{{auto}}"`? Maybe we can document this.

File: lib/redux/middleware/local_storage_middleware.dart [hunk 3]
```diff
@@ -19,9 +19,11 @@ class LocalStorageMiddleware extends MiddlewareClass<AppState> {
       final packageInfo = await PackageInfo.fromPlatform();
       final version = 'Version ${packageInfo.version} (${packageInfo.buildNumber})';
       store.dispatch(RehydrateSuccessAction(authToken, version));
+      store.dispatch(FetchAuthenticatedUserAction());
```
Comment (denrase) [inline lib/redux/middleware/local_storage_middleware.dart:22]: Do we really need to refetch the user every time the app starts up?

- User email might change, but not id?
- Doesn't the Sentry SDK persist the user info?

**Review by denrase — CHANGES_REQUESTED** [review]
Since we are now connecting personally identifiable information of users with analytics, we will need to show the appropriate privacy labels in the Apple AppStore. Also, if for some reason this info might get used for advertising or third parties, we need to present a popup, asking users for permission.

File: lib/redux/reducers.dart (hunk 1 above)
Comment (HazAT) [reply lib/redux/reducers.dart:18]: I already added this to the app privacy in itunesconnect.

File: lib/redux/middleware/sentry_sdk_middleware.dart (hunk 2 above)
Comment (HazAT) [reply lib/redux/middleware/sentry_sdk_middleware.dart]: Added a comment

**Review by denrase — APPROVED** [review]

**Pull request merged**